### PR TITLE
feat: CountDown component

### PR DESCRIPTION
**Pull request by miriamehm**
Created  `useDateTime.ts` that returns string in format DAYS:HOURS:MINUTES:SECONDS and will create a function that takes in start date and start time as props, as well as establishing a call to action

### CODE REVIEW

**Comment by JStoresund** [comment]
Please add the component in index.vue. I'm not certain how to set up the component with the futureDate and futureTime with correct format.

**Review by JStoresund — CHANGES_REQUESTED** [review]
All in all, the code looks great. I don't know how to organize the datetime strings, so I haven't been able to thoroughly test the functionality. If an instance of the component can be added to index.vue, I will do more testing.

File: components/home/CountDown.vue [hunk 1]
```diff
@@ -0,0 +1,197 @@
+<script setup lang="ts">
+  export interface CountDownContent {
+    futureDate: string
+    futureTime: string
+  }
```
[the remark this answers is not in the document]
Comment (miriamehm) [reply components/home/CountDown.vue:5]: Added the component to index.vue under HomeBanner

File: components/home/CountDown.vue [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  })
+
+  export interface TimeDiffReturn {
+    days: string
+    hours: string
+    minutes: string
+    seconds: string
+  }
+
+  const timeDiff = ref<object | null>(null)
```
Comment (AasmundN) [inline components/home/CountDown.vue:18]: Declare like this instead to properly render the page on the server.

```ts
const timeDiff = ref(
  calculateTimeDifference(
    cdProps.content.futureDate,
    cdProps.content.futureTime
  )
)
```

File: composables/useDateTime.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+export const calculateTimeDifference = (
+  futureDate: string,
+  futureTime: string
+): object => {
```
Comment (AasmundN) [inline composables/useDateTime.ts:4]: Remove `: object`. The return type will be automatically inferred.

File: components/home/CountDown.vue [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    futureDate: string
+    futureTime: string
+  }
+
+  const cdProps = defineProps({
+    content: { type: Object as PropType<CountDownContent>, required: true },
+  })
+
+  export interface TimeDiffReturn {
+    days: string
```
Comment (AasmundN) [inline components/home/CountDown.vue:12]: This can be removed after fixing the other comments.

**Review by AasmundN — CHANGES_REQUESTED** [review]
Looks great! After fixing the small, requested changes you can merge.

**Comment by AasmundN** [comment]
Also, you have to merge dev

File: pages/index.vue [hunk 5]
```diff
@@ -9,6 +9,13 @@
       }"
     />
 
+    <HomeCountDown
+      :content="{
+        futureDate: '2024-02-14',
+        futureTime: '10:00:00',
+      }"
+    ></HomeCountDown>
```
Comment (ipeglin) [inline pages/index.vue:17]: Maybe rewrite this to 
```vue
<HomeCountDown
      :content="{
        futureDate: '2024-02-14',
        futureTime: '10:00:00',
      }"
/>
```?

File: components/home/CountDown.vue [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        cdProps.content.futureDate,
+        cdProps.content.futureTime
+      )
+    }, 1000)
+  })
+</script>
+
+<template>
+  <!-- eslint-disable-next-line vue/max-len -->
+  <!-- Card containing 2 rows and a VCardAction at the bottom. First row is for event take off string, second row contains 4 cols with 2 rows each. The 2 rows in these 4 cols contains the number for days/hours/minutes/seconds and the corresponding text ("days/hours/minutes/seconds"), respectively -->
```
Comment (ipeglin) [inline components/home/CountDown.vue:32]: I think you can omit the description comment here. If we have the time in the spring, we will write documentation for the code in form of a WIKI. Hence we can include a component description there.

Optionally, you can leave a comment within the `<script></script>` in the `CountDownContent` interface to specify the format the string format should be.

Comment (AasmundN) [reply components/home/CountDown.vue:32]: Agreed

**Comment by AasmundN** [comment]
Getting an hydration mismatch...

**Pull request merged**